src/access/AccessControl.h

```diff
@@ -353,12 +354,35 @@ class AccessControl
 
     AccessControl() = default;
 
-    AccessControl(Delegate & delegate) : mDelegate(delegate) {}
-
     AccessControl(const AccessControl &) = delete;
     AccessControl & operator=(const AccessControl &) = delete;
 
-    ~AccessControl() { mDelegate.Release(); }
+    ~AccessControl() { mDelegate->Release(); }
+
+    /**
+     * @brief Set the delegate that will handle access control entries. Must be called before Init().
+     *
+     * @param delegate - The delegate to use
+     * @return CHIP_NO_ERROR on success, CHIP_ERROR_INCORRECT_STATE if set after Init(), and
+     *         CHIP_ERROR_INVALID_ARGUMENT if delegate is null.
+     */
+    CHIP_ERROR SetDelegate(Delegate * delegate)
```

Comment:
Did you consider having Init take something like `Delegate *delegateOverride=nullptr` or some InitParams that contain a `Optional<Delegate*> delegate` if we want it even clearer?

That way we guarantee: init must be called (I assume it should be) and delegate is set and used on init only and only once. Is the SetDelegate codepath generally separate than Init()?

Reply:
Reworked to remove SetDelegate